### PR TITLE
Don't fail all of an iteration of the device list retry loop on error

### PR DESCRIPTION
**Pull request by babolivier**
Without this patch, if an error happens which isn't caught by `user_device_resync`, then `_maybe_retry_device_resync` would fail, without retrying the next users in the iteration. This patch fixes this so that it now only logs an error in this case.

The reason we're logging as debug here is because if a server is consistently making this fail for a user then we'd end up flooding the logs with these errors, at a rate of one line every 30s. Though it might be fine to log it as info, and I can be convinced either way.

### CODE REVIEW

File: changelog.d/7609.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+Prevent an entire iteration of the device list resync loop to fail if one server responds with a malformed result.
```
Comment (anoadragon453) [inline changelog.d/7609.bugfix:1]:
```suggestion
Prevent an entire iteration of the device list resync loop from failing if one server responds with a malformed result.
```

File: synapse/handlers/device.py [hunk 2]
```diff
@@ -704,22 +704,33 @@ def _maybe_retry_device_resync(self):
...
-                # (through self.store.update_remote_device_list_cache).
-                if result:
+                try:
+                    # Try to resync the current user's devices list. Exception handling
+                    # isn't necessary here, since user_device_resync catches all
+                    # instances of "Exception" that might be raised from the federation
+                    # request. This means that if an exception is raised by this
+                    # function, it must be because of a database issue, which means
+                    # _maybe_retry_device_resync probably won't be able to go much
+                    # further anyway.
```
Comment (anoadragon453) [inline synapse/handlers/device.py:714]: If you need to do exception handling here, then this comment is probably a lie.

Comment (babolivier) [reply synapse/handlers/device.py:714]: Ooops yes, indeed.

Comment (babolivier) [reply synapse/handlers/device.py]: Fixed in https://github.com/matrix-org/synapse/pull/7609/commits/ace95a5d2731e7ce9f6e7b65fa917f95a42ad0f7

**Pull request merged**